lib/endpoints/class-wp-rest-users-controller.php

```diff
@@ -285,21 +288,68 @@ public function create_item( $request ) {
 				return $ret['errors'];
 			}
 		}
-
-		if ( is_multisite() ) {
-			$user_id = wpmu_create_user( $user->user_login, $user->user_pass, $user->user_email );
-			if ( ! $user_id ) {
-				return new WP_Error( 'rest_user_create', __( 'Error creating new resource.' ), array( 'status' => 500 ) );
-			}
-			$user->ID = $user_id;
-			$user_id = wp_update_user( $user );
-			if ( is_wp_error( $user_id ) ) {
-				return $user_id;
+		
+		if( empty( $request['password'] ) ){
+			$user_password = wp_generate_password( 12, false );
+		} else {
+			$user_password = $request['password'];		
+		}
+		
+		if( current_user_can( 'create_users' ) ){
+			
+			//admin mode
+			
+			if ( is_multisite() ) {
+				$user_id = wpmu_create_user( $user->user_login, $user->user_pass, $user->user_email );
+				if ( ! $user_id ) {
+					return new WP_Error( 'rest_user_create', __( 'Error creating new resource.' ), array( 'status' => 500 ) );
+				}
+				$user->ID = $user_id;
+				$user_id = wp_update_user( $user );
+				if ( is_wp_error( $user_id ) ) {
+					return $user_id;
+				}
+			} else {
+				$user_id = wp_insert_user( $user );
+				if ( is_wp_error( $user_id ) ) {
+					return $user_id;
+				}
 			}
 		} else {
-			$user_id = wp_insert_user( $user );
-			if ( is_wp_error( $user_id ) ) {
-				return $user_id;
+			
+			//front-end mode
+			
+			if ( is_multisite() ) {
+				$user_id = wpmu_create_user( $user->user_login, $user->user_pass, $user->user_email );
+				if ( ! $user_id ) {
+					return new WP_Error( 'rest_user_create', __( 'Error creating new resource.' ), array( 'status' => 500 ) );
+				} else {
+					$user->ID = $user_id;
+					$user_id = wp_update_user( $user );
+					if ( is_wp_error( $user_id ) ) {
+						return $user_id;
+					} else {
+						/**
+						  * Fires after a new user has been created via the network user-new.php page.
+						  *
+						  * @since 4.4.0
+						  *
+						  * @param int $user_id ID of the newly created user.
+						  */
+						do_action( 'network_user_new_created_user', $user_id );
+					}
+				}
+			}else{
+				$user_id = register_new_user( $user->user_login, $user->user_email );
```

Comment:
...and I guess strip_tags() is not an option,right? :)

Ok, I'll internalize the function, who can give me the error messages?